xarray/plot/plot.py

```diff
@@ -323,7 +407,15 @@ def line(darray, *args, **kwargs):
     _update_axes(ax, xincrease, yincrease, xscale, yscale,
                  xticks, yticks, xlim, ylim)
 
-    return primitive
+    if animate_over is None:
+        return primitive
+    else:
+        from animatplot.animation import Animation
+        anim = Animation([line_block, title_block], timeline=timeline)
+        # TODO I think ax should be passed to timeline_slider args
+        # but that just plots a single huge timeline and no line plot?!
+        anim.controls(timeline_slider_args={'text': animate_over})
```

Comment:
You can pass ```'valfmt':'%s'``` to format datetime. It will require some code to be smarter depending on the type of data being formated. I haven't worked with datetime much, so I'm not sure the best practices about formatting them.

Reply:
> You can pass 'valfmt':'%s' to format datetime.

This gives you the date in year-month-day hour:minute:seconds.milliseconds format, but doesn't truncate all the decimal points with the milliseconds. I also haven't used `datetime64` objects much so I don't really know how to fix that in a general manner.

> If it's in the title; maybe it doesn't need to be in the slider?

I think it should be in both, although in this example the title looks good, in my experience then once you slice high-dimensional data the title becomes really complicated and it's better to definitely have it presented nicely by the timeline too.

Reply:
I've written in a special case for datetime objects, which truncates them thanks to `pandas.to_datetime()`. Unfortunately the string still gets partly obscured by the play/pause button, but I think fixing that would require either truncating the string to a fixed maximum size or changing animatplot to choose the button placement based on the size of the string. I've updated the gif in the original PR to show what it looks like now.